server/src/main/java/com/google/musicanalysis/site/PerspectiveServlet.java

```diff
@@ -0,0 +1,31 @@
+package com.google.musicanalysis.site;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/api/perspective")
+public class PerspectiveServlet extends HttpServlet {
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse res)
+      throws ServletException, IOException {
+    PerspectiveRequest request;
```

Comment:
Instead of declaring this variable at the top of your function, just declare it when you need it:
```
var attributes = new ArrayList<String>(...);
var request = new PerspectiveRequest(..., attributes);
```

Reply:
Please use `var` when declaring variables with constructors since the variable type is obvious based on the constructor.